pilot/pkg/model/extensions.go

```diff
@@ -75,32 +80,48 @@ func convertToWasmPluginWrapper(plugin *config.Config) *WasmPluginWrapper {
 	if u.Scheme == "" {
 		u.Scheme = ociScheme
 	}
-
+	// Normalize the image pull secret to the full resource name.
+	wasmPlugin.ImagePullSecret = toSecretResourceName(wasmPlugin.ImagePullSecret, plugin.Namespace)
```

Comment:
This is mutating the object that actually lives in the k8s informer which seems problematic. In VS we do similar but make a deepcopy. Worried as-is it may lead to race or something?

Reply:
better not mutate underlying store,